docs/source/restore/index.rst

```diff
@@ -6,162 +6,79 @@ Restore
-   # cluster host IDs.
-   host_id:
-     75.237.179.49: 9b6b5206-63c8-4cdf-9b4f-23e37977fd14
-     75.196.92.133: 2d8045e7-51a4-42c4-bb8f-5e8aa77a8228
-   # destination IP: source ID
-
-The ``hosts`` file would look like:
-
-.. code::
-
-   75.237.179.49
-   75.196.92.133
-   
-
-IP to host ID mapping
-.....................
-
-SSH to one of the nodes and execute the following command:
-
-.. code-block:: none
-
-   scylla-manager-agent download-files -L <backup-location> --list-nodes
-
-it gives you information about all the clusters and nodes available in the backup location.
-
-Example output:
-
-.. code-block:: none
-
-   Cluster: prod (a9dcc6e1-17dc-4520-9e03-0a92011c823c)
-   AWS_EU_CENTRAL_1:
-   - 18.198.164.180 (7e68421b-acb1-44a7-a1a8-af7eaf1bb482)
-   - 3.122.35.120 (adc0a3ce-dade-4672-981e-26f91a3d35cb)
-   - 3.127.175.215 (2a575244-3e3c-44a1-a526-da4394f9525e)
-   - 3.65.85.108 (82f0f486-370d-4cfd-90ac-46464c8012cb)
-   - 3.66.209.145 (9ee92c19-5f78-4865-a287-980218963d96)
-   - 3.66.44.239 (aff05f79-7c69-4ecf-a827-5ea790a0fdc6)
-
-   Cluster: test (da5721cd-e2eb-4d10-a3a7-f729b8f72abf)
-   AWS_EU_CENTRAL_1:
-   - 3.123.55.215 (4001206a-3377-40cb-abd4-d38aad5dec41)
-   - 3.127.1.189 (c6466011-02f9-49dd-8951-c32028dfc6f1)
-   - 3.64.219.214 (bc39bb07-7a21-41cd-b576-51d44c1a694a)
-
-For each node IP in the new cluster you need to assign a host ID (UUID in the listing above).
-The mapping must be put into ``host_id`` variable in ``vars.yaml`` file.
-
-Snapshot tag
-............
+Process
+=======
 
-To list available snapshot tags for a node use ``--list-snapshots`` flag.
+Restore procedure works iteratively over restored locations, manifests and tables.
 
-.. code-block:: none
+    * For each backup location:
 
-   scylla-manager-agent download-files -L <backup-location> --list-snapshots -n <host-id>
+      * Await schema agreement
+      * Find live Scylla nodes that will be used for restoring current location
 
-You can filter snapshot tags containing a specific keyspaces or tables by using glob patterns.
-The parameter ``-K, --keyspace <list of glob patterns to find keyspaces>`` lets you do that.
-It works the same way as in scheduling backups or repairs with Scylla Manager.
+        * Find nodes local to location
+        * If non can be found, find any node with location access
 
-The snapshot ID must be put into ``snapshot_tag`` variable in ``vars.yaml`` file.
+      * List backup manifests for specified snapshot tag
+    * For each manifest:
 
-Inventory
-.........
+        * Filter relevant tables from the manifest
+        * For each table:
 
-Put public IP addresses of all nodes to ``hosts`` file.
+          * Temporarily disable auto-compaction of current table
+          * Temporarily disable GC grace seconds of current table
+          * Group SSTables to bundles by ID
+          * Join bundles into batches containing ``number of shards * batch size`` bundles each
```

Comment:
I don't see a reason for including `numer_of_shards` into bundle size.
See https://github.com/scylladb/scylla-manager/pull/3331#discussion_r1147207544 from one of the open PRs.

**Example:**
Let's assume this is the content of single node table backup consist of...
```
drwxr-xr-x 2 karkok karkok     4096 mar 28 12:47 .
drwxr-xr-x 3 karkok karkok     4096 mar 28 12:46 ..
-rw-r--r-- 1 karkok karkok     1090 mar 28 12:46 md-2-big-CompressionInfo.db
-rw-r--r-- 1 karkok karkok   531330 mar 28 12:46 md-2-big-Data.db
-rw-r--r-- 1 karkok karkok       10 mar 28 12:46 md-2-big-Digest.crc32
-rw-r--r-- 1 karkok karkok      176 mar 28 12:46 md-2-big-Filter.db
-rw-r--r-- 1 karkok karkok     1275 mar 28 12:46 md-2-big-Index.db
-rw-r--r-- 1 karkok karkok    89460 mar 28 12:46 md-2-big-Scylla.db
-rw-r--r-- 1 karkok karkok     4611 mar 28 12:46 md-2-big-Statistics.db
-rw-r--r-- 1 karkok karkok      344 mar 28 12:46 md-2-big-Summary.db
-rw-r--r-- 1 karkok karkok      102 mar 28 12:46 md-2-big-TOC.txt
-rw-r--r-- 1 karkok karkok     1090 mar 28 12:46 md-3-big-CompressionInfo.db
-rw-r--r-- 1 karkok karkok   531325 mar 28 12:46 md-3-big-Data.db
-rw-r--r-- 1 karkok karkok       10 mar 28 12:46 md-3-big-Digest.crc32
-rw-r--r-- 1 karkok karkok      176 mar 28 12:46 md-3-big-Filter.db
-rw-r--r-- 1 karkok karkok     1275 mar 28 12:46 md-3-big-Index.db
-rw-r--r-- 1 karkok karkok    88954 mar 28 12:46 md-3-big-Scylla.db
-rw-r--r-- 1 karkok karkok     4611 mar 28 12:46 md-3-big-Statistics.db
-rw-r--r-- 1 karkok karkok      344 mar 28 12:46 md-3-big-Summary.db
-rw-r--r-- 1 karkok karkok      102 mar 28 12:46 md-3-big-TOC.txt
-rw-r--r-- 1 karkok karkok   125954 mar 28 12:47 md-4-big-CompressionInfo.db
-rw-r--r-- 1 karkok karkok 64862613 mar 28 12:47 md-4-big-Data.db
-rw-r--r-- 1 karkok karkok       10 mar 28 12:47 md-4-big-Digest.crc32
-rw-r--r-- 1 karkok karkok    19544 mar 28 12:47 md-4-big-Filter.db
-rw-r--r-- 1 karkok karkok   171328 mar 28 12:47 md-4-big-Index.db
-rw-r--r-- 1 karkok karkok    90296 mar 28 12:47 md-4-big-Scylla.db
-rw-r--r-- 1 karkok karkok     4611 mar 28 12:47 md-4-big-Statistics.db
-rw-r--r-- 1 karkok karkok    37112 mar 28 12:47 md-4-big-Summary.db
-rw-r--r-- 1 karkok karkok      102 mar 28 12:47 md-4-big-TOC.txt
-rw-r--r-- 1 karkok karkok   127874 mar 28 12:47 md-5-big-CompressionInfo.db
-rw-r--r-- 1 karkok karkok 65854908 mar 28 12:47 md-5-big-Data.db
-rw-r--r-- 1 karkok karkok       10 mar 28 12:47 md-5-big-Digest.crc32
-rw-r--r-- 1 karkok karkok    19840 mar 28 12:47 md-5-big-Filter.db
-rw-r--r-- 1 karkok karkok   173957 mar 28 12:47 md-5-big-Index.db
-rw-r--r-- 1 karkok karkok    90296 mar 28 12:47 md-5-big-Scylla.db
-rw-r--r-- 1 karkok karkok     4611 mar 28 12:47 md-5-big-Statistics.db
-rw-r--r-- 1 karkok karkok    37672 mar 28 12:47 md-5-big-Summary.db
-rw-r--r-- 1 karkok karkok      102 mar 28 12:47 md-5-big-TOC.txt
-rw-r--r-- 1 karkok karkok    91586 mar 28 12:47 md-6-big-CompressionInfo.db
-rw-r--r-- 1 karkok karkok 47158286 mar 28 12:47 md-6-big-Data.db
-rw-r--r-- 1 karkok karkok       10 mar 28 12:47 md-6-big-Digest.crc32
-rw-r--r-- 1 karkok karkok    14208 mar 28 12:47 md-6-big-Filter.db
-rw-r--r-- 1 karkok karkok   124424 mar 28 12:47 md-6-big-Index.db
-rw-r--r-- 1 karkok karkok    90296 mar 28 12:47 md-6-big-Scylla.db
-rw-r--r-- 1 karkok karkok     4611 mar 28 12:47 md-6-big-Statistics.db
-rw-r--r-- 1 karkok karkok    26984 mar 28 12:47 md-6-big-Summary.db
-rw-r--r-- 1 karkok karkok      102 mar 28 12:47 md-6-big-TOC.txt
-rw-r--r-- 1 karkok karkok    88050 mar 28 12:47 md-7-big-CompressionInfo.db
-rw-r--r-- 1 karkok karkok 45335526 mar 28 12:47 md-7-big-Data.db
-rw-r--r-- 1 karkok karkok       10 mar 28 12:47 md-7-big-Digest.crc32
-rw-r--r-- 1 karkok karkok    13664 mar 28 12:47 md-7-big-Filter.db
-rw-r--r-- 1 karkok karkok   119595 mar 28 12:47 md-7-big-Index.db
-rw-r--r-- 1 karkok karkok    90296 mar 28 12:47 md-7-big-Scylla.db
-rw-r--r-- 1 karkok karkok     4611 mar 28 12:47 md-7-big-Statistics.db
-rw-r--r-- 1 karkok karkok    25944 mar 28 12:47 md-7-big-Summary.db
-rw-r--r-- 1 karkok karkok      102 mar 28 12:47 md-7-big-TOC.txt
```

We have 42 SSTables here.
Setup of the destination cluster is 5 nodes 32 shards each.
Default batch_size is 2.
It means that single bundle may contain 64 SSTables.

==> it generally means that only one node will be participating in the restore, because it will pickup all table sstables.
4 other nodes will be in the `idle` state.